### ISSUE TITLE
10$: improve error for 'type \n T<U> = U;'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#846]
The following code is invalid TypeScript. quick-lint-js reports a bad diagnostic:
```typescript
type
  T<U> = U;
```
We should report E0277 for this case.

Similar diagnostics has better error recovery: E0275, E0276

**Comment by cdvalley** [comment]
Hi, I would like to take this issue if possible.

As required: I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by strager** [comment]
@cdvalley Sure. Let me know if you need any help.
